packages/react-compass/src/button/button.styles.ts

```diff
@@ -202,6 +202,58 @@ export const StyledButton = styled('button', {
           background: '$cdgBlue10',
         },
       },
+      'h5-primary': {
+        background: '$blueShades100',
+        color: '$whiteText',
+        borderColor: '$grayShades20',
+        '&.cdg-link-button:hover': {
+          background: '$blueShades120',
+          borderColor: '$grayShades20',
+        },
+        '&:hover:enabled': {
+          background: '$blueShades120',
+          borderColor: '$grayShades20',
+        },
+      },
+      'h5-secondary': {
+        background: 'transparent',
+        color: '$blueShades100',
+        borderColor: '$blueShades100',
+        '&.cdg-link-button:hover': {
+          color: '$blueShades120',
+          background: '$grayShades10',
+          borderColor: '$blueShades120',
+        },
+        '&:hover:enabled': {
+          color: '$blueShades120',
+          background: '$grayShades10',
+          borderColor: '$blueShades120',
+        },
+      },
+      'h5-danger': {
+        background: 'transparent',
+        color: '$dangerShades',
+        borderColor: '$dangerShades',
+        '&.cdg-link-button:hover': {
+          background: '$grayShades10',
+        },
+        '&:hover:enabled': {
+          background: '$grayShades10',
+        },
+      },
+    },
+    h5: {
+      true: {
+        borderRadius: '$lg',
+        padding: '11px $6',
+        fontSize: '$body2',
+        fontWeight: '$bold',
+        '&:disabled': {
+          background: '$grayShades20',
+          color: '$grayShades60',
+          borderColor: '$grayShades20',
+        },
+      },
```

Comment:
I suggest putting the style of `h5-primary`,`h5-seconady`,`h5-danger` to compound variant, for example:
![image](https://github.com/comfortdelgro/compass-design/assets/7071853/2f629296-3d66-4d1d-a52b-7792145b8a8e)

By doing that we don't need 3 more value for the props `variant` anymore.
![image](https://github.com/comfortdelgro/compass-design/assets/7071853/fcac4f7d-9b35-4cf1-93ba-cd9b46de3257)

Reply:
updated, thanks @pieliedie